### PR TITLE
Optional manager

### PR DESCRIPTION
**Pull request by mcritchlow**
References #369 

#### Local Checklist
- [x] Tests written and passing locally?
- [x] Code style checked?
- [x] QA-ed locally?
- [x] Rebased with `master` branch?
- [x] Configuration updated (if needed)?
- [ ] Documentation updated (if needed)?

#### What does this PR do?
- Adds missing `pry-byebug` gem for debugging
- Allow `Employee` records to optionally have a `manager` instead of requiring it.

We have three scenarios that have been identified that could cause an
`Employee`, that is valid as far as the application is concerned, to not
have a `manager` attribute in LDAP

1. The `ldap_sync` script simply has not run and updated the employee record. This could happen for new employees or employees who change jobs in the library
2. The employee has a dual-assignment, and their AD record is not owned and managed by the Library
3. The employee has a dual-assignment, and their AD record is owned and managed by the Library, but their manager does not work for the Library

##### Why are we doing this? Any context of related work?
References #369 

The application needs to be made a bit more flexible to accommodate some edge cases. Shocking, we know.

#### Where should a reviewer start?
Please take a hard look at this. It's a fairly large change

- Look at the migration and resulting `db/schema`
- Look at the new test coverage. I tried to cover the scenarios I could think of. Am I missing anything?
- Other concerns or additional edge cases I haven't covered here?

#### Database changes
Change the `manager` column in the `employees` table to allow `NULL` values.

@ucsdlib/developers - please review

### CODE REVIEW

File: spec/mailers/recognition_mailer_spec.rb [hunk 1]
```diff
@@ -3,53 +3,82 @@
...
-  end
+  context 'with a valid manager' do
+    let!(:manager) { FactoryBot.create(:employee) }
+    let(:recognition) { FactoryBot.build_stubbed(:recognition,
+                                                   user: user,
+                                                   created_at: Time.parse('2018-10-01'),
+                                                   description: 'in report',
+                                                   employee: employee) }
+    let(:employee) { FactoryBot.create(:employee, manager: manager.uid) }
+    let(:employee_without_manager) { FactoryBot.create(:employee_without_manager) }
```
Comment (VivianChu) [inline spec/mailers/recognition_mailer_spec.rb]: @mcritchlow - It seems like this variable employee_without_manager is not in used in this context.

Comment (mcritchlow) [reply spec/mailers/recognition_mailer_spec.rb]: Oh good catch, thank you! I think I had initially started here, and then moved to a dedicated context. I'll remove that.
